src/handlers/tests/interventions-processor-lambda.test.ts

```diff
@@ -405,5 +464,89 @@ describe('intervention processor handler', () => {
       expect(logAndPublishMetric).toHaveBeenCalledWith('CONFIDENCE_LEVEL_TOO_LOW');
       expect(logger.warn).toHaveBeenCalledWith('Received interventions has low level of confidence: P1');
     });
+
+    it('It should log an error and continue if the inner message body is not valid JSON', async () => {
+      mockRecord = {
+        messageId: '123',
+        receiptHandle: '',
+        body: JSON.stringify({ notTheRightStructure: 'something', body: 'not a JSON string'}),
+        attributes: {
+          ApproximateReceiveCount: '',
+          SentTimestamp: '',
+          SenderId: '',
+          ApproximateFirstReceiveTimestamp: '',
+        },
+        messageAttributes: {},
+        md5OfBody: '',
+        eventSource: '',
+        eventSourceARN: '',
+        awsRegion: '',
+      };
+      expect(await handler({ Records: [mockRecord] }, mockContext)).toEqual({
+        batchItemFailures: [],
+      });
+      expect(logger.error).toHaveBeenCalledWith('Message body is not valid JSON.');
+      expect(logAndPublishMetric).toHaveBeenCalledWith(MetricNames.INVALID_EVENT_RECEIVED);
+
+    });
+
+    it('If a valid and invalid events are received none should be retried', async () => {
```

Comment:
```suggestion
    it('should be not retry, if both valid and invalid events are received', async () => {
```

Reply:
sorry, I don't think the suggestions reads correctly